src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -0,0 +1,794 @@
+        /// into all its individual datetime components. Input column must be of type Int64 representing the number of seconds since the unix epoc.
+        /// This transformer will append the <paramref name="columnPrefix"/> to all the output columns.
+        /// </summary>
+        /// <param name="catalog">Transform catalog</param>
+        /// <param name="inputColumnName">Input column name</param>
+        /// <param name="columnPrefix">Prefix to add to the generated columns</param>
+        /// <returns><see cref="DateTimeEstimator"/></returns>
+        public static DateTimeEstimator DateTimeTransformer(this TransformsCatalog catalog, string inputColumnName, string columnPrefix)
+            => new DateTimeEstimator(CatalogUtils.GetEnvironment(catalog), inputColumnName, columnPrefix);
+
+        /// <summary>
+        /// Create a <see cref="DateTimeEstimator"/>, which splits up the input column specified by <paramref name="inputColumnName"/>
+        /// into all its individual datetime components. Input column must be of type Int64 representing the number of seconds since the unix epoc.
+        /// This transformer will append the <paramref name="columnPrefix"/> to all the output columns. If you specify a country,
+        /// Holiday details will be looked up for that country as well.
+        /// </summary>
+        /// <param name="catalog">Transform catalog</param>
+        /// <param name="inputColumnName">Input column name</param>
+        /// <param name="columnPrefix">Prefix to add to the generated columns</param>
+        /// <param name="country">Country name to get holiday details for</param>
+        /// <returns><see cref="DateTimeEstimator"/></returns>
+        public static DateTimeEstimator DateTimeTransformer(this TransformsCatalog catalog, string inputColumnName, string columnPrefix, DateTimeEstimator.HolidayList country = DateTimeEstimator.HolidayList.None)
+            => new DateTimeEstimator(CatalogUtils.GetEnvironment(catalog), inputColumnName, columnPrefix, country);
+
+        #region ColumnsProduced static extentions
+
+        internal static Type GetRawColumnType(this DateTimeEstimator.ColumnsProduced column)
+        {
+            switch (column)
+            {
+                case DateTimeEstimator.ColumnsProduced.Year:
+                case DateTimeEstimator.ColumnsProduced.YearIso:
+                    return typeof(int);
+                case DateTimeEstimator.ColumnsProduced.DayOfYear:
+                case DateTimeEstimator.ColumnsProduced.WeekOfMonth:
+                    return typeof(ushort);
+                case DateTimeEstimator.ColumnsProduced.MonthLabel:
+                case DateTimeEstimator.ColumnsProduced.AmPmLabel:
+                case DateTimeEstimator.ColumnsProduced.DayOfWeekLabel:
+                case DateTimeEstimator.ColumnsProduced.HolidayName:
```

Comment:
Usability question, why are we producing multiple output columns of instead of a single vector type columns?

The next step after this DateTime transform is:
* **Convert** all the INTs/USHORTs/BYTEs columns to Float32
* **One-hot encode** the STRING columns (`"Friday"` to `[ 0, 0,  0, 0,  0, 1, 0 ]`)
* **Concatenate** all of these feature to single vector of Float32

Why aren't we just outputting the final vector type column? Then the user can directly use its output, like our other transforms.

Reply:
If there's a usecase for the individual columns, perhaps just a boolean parameter `produceRawColumns`, which then produces the individual outputs instead of a singular vector column?

Reply:
We did it this way just because thats how the original python code works. It splits everything up into different columns and doesn't combine them into a vector.

Reply:
Having a single vector is consistent with `FeaturizeText`.

Reply:
The original code we were bringing to ML.NET returned all individual columns. Because of that, for now, we will be leaving them as individual columns. If we learn that we never need individual columns and only use this as a vector then we can revisit this then.